src/io/flutter/run/daemon/DaemonApi.java

```diff
@@ -236,6 +203,64 @@ private Command takePending(int id) {
     }
   }
 
+  /**
+   * Parse the given string; if it's valid JSON, and a valid Daemon response of event, then return
+   * the parsed JsonObject.
+   */
+  static JsonObject parseAndValidateDaemonEvent(String str) {
+    if (!str.startsWith("[{")) {
+      return null;
+    }
+
+    str = str.trim();
```

Comment:
Should the trim come first?

Reply:
good question - in this case we do know that we won't have leading ws